### PR TITLE
[core] Remove backports of C++17 features

### PR DESCRIPTION
**Pull request by guitargeek**
Remove the `std::string_view` backport and others, because it's not necessary anymore since the minimum C++ standard is now C++17.

### CODE REVIEW

**Comment by vgvassilev** [comment]
Thank you for cleaning up that mess!

File: core/foundation/inc/ROOT/RWrap_libcpp_string_view.h [hunk 1]
```diff
@@ -1,591 +0,0 @@
-// -*- C++ -*-
```
Comment (Axel-Naumann) [inline core/foundation/inc/ROOT/RWrap_libcpp_string_view.h:1]: GitHub tells me that this file got emptied but isn't deleted. Is that intentional? Or a bug in GitHub's interface?

File: core/foundation/inc/ROOT/libcpp_string_view.h [hunk 2]
```diff
@@ -1,824 +0,0 @@
-// -*- C++ -*-
```
Comment (Axel-Naumann) [inline core/foundation/inc/ROOT/libcpp_string_view.h:1]: Same here: empty but file remains?

File: core/base/inc/TString.h [hunk 3]
```diff
@@ -309,15 +309,8 @@ friend std::strong_ordering operator<=>(const TString &s1, const TString &s2) {
 
    // Type conversion
    operator const char*() const { return GetPointer(); }
-#if (__cplusplus >= 201700L) && !defined(_MSC_VER) && (!defined(__clang_major__) || __clang_major__ > 5)
```
Comment (Axel-Naumann) [inline core/base/inc/TString.h:312]: How certain are we that Windows doesn't need this anymore? I guess the CI will tell...

File: tmva/tmva/inc/TMVA/DNN/Architectures/Cuda/CudaMatrix.h [hunk 4]
```diff
@@ -22,12 +22,6 @@
 // in case we compile C++ code with std-17 and cuda with lower standard
 // use experimental string_view, otherwise keep as is
 #include "RConfigure.h"
-#ifdef R__HAS_STD_STRING_VIEW
-#ifndef R__CUDA_HAS_STD_STRING_VIEW
-#undef R__HAS_STD_STRING_VIEW
-#define R__HAS_STD_EXPERIMENTAL_STRING_VIEW
```
Comment (Axel-Naumann) [inline tmva/tmva/inc/TMVA/DNN/Architectures/Cuda/CudaMatrix.h:28]: Isn't anything using these preprocessor defines?

**Review by Axel-Naumann — APPROVED** [review]
Excellent, thanks for doing this! See comments, please! And let's make sure that the CI passes before merging :-)

File: core/foundation/inc/ROOT/RWrap_libcpp_string_view.h (hunk 1 above)
Comment (guitargeek) [reply core/foundation/inc/ROOT/RWrap_libcpp_string_view.h:1]: gh_iface_bug

For me it shows it like this, so correctly deleted. Maybe a bug in the comparison interface

File: core/base/inc/TString.h (hunk 3 above)
Comment (guitargeek) [reply core/base/inc/TString.h:312]: Yes :)

File: tmva/tmva/inc/TMVA/DNN/Architectures/Cuda/CudaMatrix.h (hunk 4 above)
Comment (guitargeek) [reply tmva/tmva/inc/TMVA/DNN/Architectures/Cuda/CudaMatrix.h:28]: My gut feeling here is "pretty sure", but maybe @lmoneta can also comment

**Comment by guitargeek** [comment]
Thanks @Axel-Naumann for the green light! FYI, I have added one more commit that replaces the usage of the `RStringView` header with `<string_view>`.

Also, can you please approve the sister PR in roottest to fix the CI failures? Thanks!
https://github.com/root-project/roottest/pull/1031

**Comment by guitargeek** [comment]
Oh, and the Windows 10 failures seem unrelated to this PR, right @bellenot? In any case, this PR should *not* be merged before we see any green Windows build

**Comment by bellenot** [comment]
@guitargeek the only failure I see (on Jenkins) was due to `Out of diskspace` error. But it's running on another node now, so we'll see. (and I'm cleaning up the Windows nodes right now)

File: core/base/inc/TString.h (hunk 3 above)
Comment (guitargeek) [reply core/base/inc/TString.h:312]: Okay, it's still needed. I don't like that it's there, but that would be another story

File: tmva/tmva/inc/TMVA/DNN/Architectures/Cuda/CudaMatrix.h (hunk 4 above)
Comment (lmoneta) [reply tmva/tmva/inc/TMVA/DNN/Architectures/Cuda/CudaMatrix.h:28]: I think these are not needed anymore. There were there to fix  conflict happening in past  Cuda version when including string_view backport

**Review by lmoneta — APPROVED** [review]

**Comment by guitargeek** [comment]
Windows is green now

File: tmva/tmva/inc/TMVA/DNN/Architectures/Cuda/CudaMatrix.h (hunk 4 above)
Comment (pcanal) [reply tmva/tmva/inc/TMVA/DNN/Architectures/Cuda/CudaMatrix.h:28]: Do we have any CUDA build in the CI?

Comment (Axel-Naumann) [reply tmva/tmva/inc/TMVA/DNN/Architectures/Cuda/CudaMatrix.h:28]: @lmoneta ?

Comment (lmoneta) [reply tmva/tmva/inc/TMVA/DNN/Architectures/Cuda/CudaMatrix.h:28]: No, we don't have it!

**Pull request merged**